src/core/channels/channel.ts

```diff
@@ -55,35 +56,42 @@ export default class Channel extends EventsDispatcher {
     this.subscriptionPending = false;
   }
 
-  /** Handles an event. For internal use only.
+  /** Handles a message. For internal use only.
    *
-   * @param {String} event
-   * @param {*} data
+   * @param {Message} message
    */
-  handleEvent(event : string, data : any) {
-    if (event.indexOf("pusher_internal:") === 0) {
-      if (event === "pusher_internal:subscription_succeeded") {
-        this.subscriptionPending = false;
-        this.subscribed = true;
-        if (this.subscriptionCancelled) {
-          this.pusher.unsubscribe(this.name);
-        } else {
-          this.emit("pusher:subscription_succeeded", data);
-        }
-      }
-    } else {
+  handleMessage(message: Message) {
```

Comment:
I'm not sure about the change of name and signature:

* I think "Event" is actually a better name than "Message" because it's what they are referred to in our docs: https://pusher.com/docs/pusher_protocol#events. I think it's the `Message` type that is poorly named.
* `Message` defines the wire format of Pusher events, and I'm concerned about it bubbling so far up the call stack. I would prefer these `handleMessage` methods to be called with only the parameters they need, e.g. `handleMessage` for public channels does not need to be called with a `user_id`. It also feels wrong that non pusher protocol events are being wrapped in a `Message`, e.g. https://github.com/pusher/pusher-js/pull/323/files#diff-0e82fa5891d842a9141e2aad30757dd7R90